pkg/v1/providers/infrastructure-vsphere/v1.3.1/ytt/overlay.yaml

```diff
@@ -69,6 +38,41 @@
 #@   return 0
 #@ end
 
+
+#@ valid_pci_devices = [[0x10DE,0x1EB8]]
+
+#@ def valid_pci_devices_list():
+#@       return ",".join(map(lambda x:  "0x%X:0x%X" % (x[0], x[1]), valid_pci_devices))
+#@ end
+
+#@ def get_pci_devices(pci_devices_string, pci_ignore_pci_device_validation):
+#@     pci_devices = list()
+#@     help_error = "VSPHERE_CONTROL_PLANE_PCI_DEVICES \"" + pci_devices_string + "\" must be in the  \"0x<vendorId>:0x<deviceId>,0x<vendorId>:0x<deviceId>,..\" format."
+#@     for val in pci_devices_string.split(","):
+#@         if len(val) <= 0:
+#@             assert.fail(help_error)
+#@         end
+#@         kv = val.split(":")
+#@         if len(kv) != 2:
+#@             assert.fail(help_error)
+#@         end
+#@         id_pair = dict()
+#@         vendor_id = int(kv[0], 16)
+#@         device_id = int(kv[1], 16)
+#@         if not pci_ignore_pci_device_validation:
+#@             for x in valid_pci_devices:
+#@                 if vendor_id != x[0] or device_id != x[1]:
+#@                     assert.fail("Device 0x%X:0x%X is not supported. Supported devices are: %s. Set VSPHERE_IGNORE_PCI_DEVICES_ALLOW_LIST to ignore this check." % (vendor_id, device_id, valid_pci_devices_list()))
+#@                 end
+#@             end
+#@         end
```

Comment:
Shouldn't the logic be like this?

```
#@        if not pci_ignore_pci_device_validation:
#@.            matchesAny = false
#@             for x in valid_pci_devices:
#@                 if vendor_id == x[0] or device_id == x[1]:
#@                     matchesAny = true
#@                 end
#@             end
#@             if not matchesAny:
#@                 assert.fail("Device 0x%X:0x%X is not supported. Supported devices are: %s. Set VSPHERE_IGNORE_PCI_DEVICES_ALLOW_LIST to ignore this check." % (vendor_id, device_id, valid_pci_devices_list()))
#@             end
#@         end
```

Reply:
Apart from `if vendor_id == x[0] or device_id == x[1]:`
needing to be `if vendor_id == x[0] and device_id == x[1]:`, and for which i've now added a test case just now, yup. Changed.